### DISCUSSION TITLE
Track mit Wegpunkten mit Text

### DISCUSSION THREAD

**Discussion by Biker729** [devemux86/cruiser#531] [Q&A] [answered]
Eine Route mit Wegpunkten mit Text von einer kurviger *.datei wird der Text von WP in Cruiser angezeigt.!

Ein Track mit Text in Wegpunkten von Basecamp wird der Text  in Cruiser nicht angezeigt.?

Ist es möglich sich einen Track mit Wegpunkten mit Text in einer GPX Datei 
anzeigen zu lassen. ?

....können die Shaping-Points ausgeblendet werden. ?

Gruß und Danke

**Comment by devemux86** [comment]
Hi @Biker729,

Waypoints and track are different things in the GPX file.
What do you select during GPX import? Waypoints or Track?
Do you have a GPX example so that we can better understand what you mean?

Wegpunkte und Track sind unterschiedliche Dinge in der GPX-Datei.
Was wählen Sie beim GPX-Import aus: Wegpunkte oder Track?
Haben Sie ein GPX-Beispiel, damit wir besser verstehen können, was Sie meinen?

**Comment by Biker729** [comment]
Wegpunkte und Track in einer GPX-Datei mit Basecamp exportieren 
und mit Cruiser Wegpunkte und Track in einer GPX-Datei importieren.

<img width="652" alt="1" src="https://github.com/devemux86/cruiser/assets/146021011/ed5d78e8-0d98-465a-bc65-ea90ee8d7c98">

In OsmAnd sieht das so aus.

![3](https://github.com/devemux86/cruiser/assets/146021011/090c2a6e-7dfb-4f7f-bc73-a4a5ac7468ba)

**Reply by devemux86** [reply]
@Biker729 

Have you selected "Waypoints" in the import dialog? Then the route will contain the waypoints with their names.
If you select "Automatic" or "Track", the route will be calculated with new waypoints.

Haben Sie im Importdialog "Wegpunkte" ausgewählt? Dann wird die Route die Wegpunkte mit ihren Namen enthalten.
Wenn Sie "Automatisch" oder "Verfolgen" wählen, wird die Route mit neuen Wegpunkten berechnet.

<img src="https://github.com/devemux86/cruiser/assets/3484020/a1bcf9f9-2cd9-4848-8985-1378c4e6d212" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/046c2ba9-b6a7-48f5-9837-8a11859a9a9f" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c7d47d8b-7160-49f1-af81-16b96489918a" width="33%"/>

**Comment by Biker729** [comment]
OK mit  „“Importdialog „Wegpunkte““ wird der Text angezeigt 
aber die Route wird nicht korrekt angezeigt.

Kann der Track nicht 1:1 angezeigt werden. ?

**Reply by devemux86** [reply]
@Biker729 

To display the track 1:1, you must select "Automatic" or "Track" in the import dialog, but then new waypoints will be created.
GPX tracks do not contain route details, so the routes must be recalculated with the selected router.

Um den Track 1:1 darzustellen, müssen Sie im Importdialog "Automatisch" oder "Track" auswählen, aber dann werden neue Wegpunkte erstellt.
GPX-Tracks enthalten keine Routendetails, so dass die Routen mit dem ausgewählten Router neu berechnet werden müssen.

**Reply by devemux86** [reply]
@Biker729 

You can import the waypoints as favorites if you need them on the map?

Sie können die Wegpunkte als Favoriten importieren, wenn Sie sie auf der Karte benötigen?

<img src="https://github.com/devemux86/cruiser/assets/3484020/c3e5c063-a698-4098-b158-5462efe7979a" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/546a8a6c-7531-49a6-b8b1-4bc60634db0b" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/f409ece2-d1c3-4915-89ee-b4d1e4870344" width="33%"/>

**Reply by WolfgangZ42** [reply, marked as answer]
Beim Import "Track" angeben, dann wird eine korrekte Route entwickelt (an einigen Stellen vielleicht noch zu korrigieren) - die Wegpunkte werden ebenfalls mit übernommen und als blaue Punkte angezeigt. Alles was bei den Wegpunkten im "Name"-Feld steht wird dann angezeigt:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/2928b638-cc8f-4662-8814-b757e5ed4732"/>

When importing, enter "Track", then a correct route will be developed (may need to be corrected in some places) - the waypoints will also be transferred and displayed as blue dots. Everything in the "Name" field of the waypoints is then displayed.

**Comment by Biker729** [comment]
@WolfgangZ42
Ja das wäre eine Möglichkeit aber jedesmal nacharbeiten ist mir zu umständlich,

Wenn ein GPX Track mit WP in einer Datei mit OsmAnd möglich ist sollte es doch mit Cruiser auch möglich sein. ?

**Reply by devemux86** [reply]
@Biker729 

If you follow @WolfgangZ42's options above, the result will be the same. What difference do you see?
OsmAnd only stores the imported waypoints in its database. I can also implement it in Cruiser if you need it.

Wenn Sie die oben genannten Optionen von @WolfgangZ42 befolgen, dann ist das Ergebnis dasselbe. Welchen Unterschied sehen Sie?
OsmAnd speichert nur die importierten Wegpunkte in seiner Datenbank. Ich kann es auch in Cruiser implementieren, wenn Sie es brauchen.

**Reply by WolfgangZ42** [reply]
> jedesmal nacharbeiten ist mir zu umständlich

Das Nacharbeiten bezog sich ja nur auf die aus dem Track errechnete Route. Wenn ich statt 50 nun 100 Wegpunkte generieren lassen, ist die Route nahezu perfekt erstellt - lediglich 1 shaping point muss noch gesetzt werden. 
Die im GPX-File enthaltenen Wegpunkte müssen überhaupt nicht bearbeitet werden und stehen komplett zur Verfügung. 

The reworking only related to the route calculated from the track. If I now generate 100 waypoints instead of 50, the route is almost perfectly created - only 1 shaping point still needs to be set. 
The waypoints contained in the GPX file do not have to be edited at all and are completely available.

**Reply by WolfgangZ42** [reply]
> Die im GPX-File enthaltenen Wegpunkte müssen überhaupt nicht bearbeitet werden und stehen komplett zur Verfügung.

Sorry - ich muss mich korrigieren: die blauen Punkte sind nach Beendigung der App weg und werden auch nicht exportiert. Man kann sie jedoch (manuel!) zu Via Punkten umwandeln und dadurch erhalten. Das GPX-File ist vermutlich aus einem Track entstanden, zu dem mehr oder weniger willkürlich Wegpunkte hinzugefügt wurden. Aus den Wegpunkten kann keine 1:1-Route erstellt werden, aus dem Track sehr wohl.  
Ich habe die Umwandlung der blauen Punkte mal gemacht - hat 4 Minuten gedauert, gestoppt! Hier das Ergebnis: [Route Des Cretes (2).gpx.txt](https://github.com/devemux86/cruiser/files/15250889/Route.Des.Cretes.2.gpx.txt)

**Comment by Biker729** [comment]
…jedenfalls muss man kontrollieren ob der Track 1:1 ist, was ein Track eigentlich sollte.

Kurviger Dateien bearbeite ich mit GpxShaping zu Track mit WP für OsmAnd.

Da ich die Vorzüge von OsmAnd Track mit WP 1:1 (ohne Nacharbeit und in einer GPX Datei) inzwischen  
sehr zu schätzen weiß möchte ich darauf nicht mehr verzichten.
 
Auch die Vorzüge mit einer App alles zu machen ob TET Track, Route, Kurviger leider nur über GpxShaping.

Da sich Cruiser sehr gut weiterentwickelt hat wäre es eine Option für mich nach Wechsel von Kurviger zu OsmAnd 
wieder zu  @devemux86  zu wechseln, da Kurviger Dateien in Gegensatz zu OsmAnd möglich sind.

Gruß und Danke

**Reply by WolfgangZ42** [reply]
> Kurviger leider nur über GpxShaping

Habe ich bisher auch immer gemacht - nur für meine Freunde, die sich partu nicht vom BMW-Navigator trennen wollen. Ich selbst kommen ja mit "native" Kurviger Dateien und Cruiser bestens zurecht. Seit kurzem ist GPXshaping aber obsolet, denn die zusätzlichen Wegpunkte werden von Cruiser nun eingefügt. Ein Tool weniger! 

Auch ich plane (und fahre) ACT und TET. Mit Kurviger nur schwer zu planen, da viele mit "track" definierte Abschnitte nicht genommen werden, mit Cruiser (im Profile "Mountainbike") viel einfacher. 


I've always done this - only for my friends who don't want to part with the BMW Navigator. I myself get along fine with "native" Kurviger files and Cruiser. Recently, however, GPXshaping has become obsolete, because the additional waypoints are now inserted by Cruiser. One tool less! 

I also plan (and ride) ACT and TET. It's difficult to plan with Kurviger, as many sections defined with "track" are not taken, but much easier with Cruiser (in the "Mountainbike" profile).

**Comment by devemux86** [comment]
@Biker729 @WolfgangZ42 

The route must be created from the track if we want a 1:1 conversion.
The gpx waypoints are not connected to the route, they are like points of interest in the gpx.
If you all want such gpx waypoints to be saved like favorites, I can implement it.

Die Route muss aus dem Track erstellt werden, wenn wir eine 1:1-Konvertierung wünschen.
Die gpx-Wegpunkte sind nicht mit der Route verbunden, sie sind wie Points of Interest in der gpx.
Wenn ihr alle wollt, dass solche gpx-Wegpunkte wie Favoriten gespeichert werden, kann ich es implementieren.

**Reply by WolfgangZ42** [reply]
Als Favoriten würde ich sie gar nicht haben wollen, denn sie gehören ja definitiv zu diesem Track. Aber als Via Point wären sie von großem Nutzen - also meine manuelle Umwandlung in automatischer Form.
Natürlich sollten sie dazu auch direkt auf der Route liegen, denn ich vielen GPX-Dateien habe ich WPT-Einträge gesehen, die (wie Favoriten) abseits der Route definiert sind.

I wouldn't want to have them as favorites, because they definitely belong to this track. But they would be very useful as via points - i.e. my manual conversion in automatic form.
Of course, they should also be directly on the route, because I have seen WPT entries in many GPX files that are defined off the route (like favorites).

**Reply by devemux86** [reply]
@WolfgangZ42 

They are not real route waypoints and the route is not created by them, so they are not part of the route.
OsmAnd also treats them separately, like points of interest. If we want them, we should save them separately.

Sie sind keine echten Routen-Wegpunkte und die Route wird nicht durch sie erstellt, also gehören sie nicht zur Route.
OsmAnd behandelt sie auch separat, wie Points of Interest. Wenn wir sie wollen, sollten wir sie separat speichern.

**Reply by devemux86** [reply]
In this case we need to adapt the offline route formats of Cruiser to support the storage of extra points of interest.

**Comment by Biker729** [comment]
@devemux86
“Die GPX-Wegpunkte sind nicht mit der Route verbunden ….“

Es wäre aber von Vorteil wenn Track und WP in der GPX gespeichert bleiben 
sonst müssen diese immer zusätzlich zum Track gelöscht werden.

![Track mit Wegpunkten in einer GPX-Datei](https://github.com/devemux86/cruiser/assets/146021011/7a94f575-426f-41c5-b20d-419d0f5e3e6f)

![Track with waypoints in a GPX file](https://github.com/devemux86/cruiser/assets/146021011/ba7a6fcd-e981-46fa-8118-944f6030dccf)

**Comment by devemux86** [comment]
How do you suggest we handle GPX files containing waypoints + track?

We can select "Overlay + Routing" on import and use the track to calculate a route similar to the track.
While the GPX waypoints are also imported as overlay of blue dots with information available with tap.
(like in the [image](https://github.com/devemux86/cruiser/discussions/531#discussioncomment-9355426))

GPX files with waypoints can also be imported as favorites in Cruiser and then _used in navigation_.

@Biker729 mentions [above](https://github.com/devemux86/cruiser/discussions/531#discussioncomment-9357218) how OsmAnd handles such GPX files with waypoints + track.
(waypoints are used in navigation)

How can we improve the "Routing | Import" in Cruiser of GPX files with waypoints + track?

**Reply by devemux86** [reply]
One idea could be that when we import GPX with waypoints + track and use the track for routing,
then the waypoints are also imported as a special favorites group that is enabled in navigation.

**Reply by MotoUKRider** [reply]
@devemux86 I don't use Tracks, but if this helps, this is what ITNConv sees when I open a GPX that I export from BaseCamp:

![itnconv-gpx-open](https://github.com/devemux86/cruiser/assets/125696109/54660b73-c654-4d6b-8e36-469b829f5885)

Cruiser has always been able to "handle" these files *perfectly*, and that was my initial draw to the app.  In my case, the 15 Waypoints in that file will already exist across several Groups (there could be 12 Photo stops and 3 Fuel breaks).  Of course, 15 of those 48 points in the Route will be the Waypoints.  ITNConv obviously lets you do a little bit more with a GPX file and allows you to extract items for conversion, but that's how it shows a "complicated" GPX.

It also handles a GPX with several Routes and allows you to load a specific one ([my earlier topic here](https://github.com/devemux86/cruiser/discussions/44)).  I currently export 1 Route per GPX file although a multi-route GPX isn't a requirement, but seems to be part of the "spec".

I suppose what I'm saying is that I don't want to end up with a load of duplicate Favorites each time I load a GPX.  An option to include them in that special group via a toggle, perhaps?

**Reply by devemux86** [reply]
> It also handles a GPX with several Routes and allows you to load a specific one (https://github.com/devemux86/cruiser/discussions/44)

This is a different feature and is already in my list to implement. :slightly_smiling_face: 

---

The most flexible way is to _import GPX waypoints as favorites_ with custom icons, colors, visibility and use in navigation.

But users (as in this report) may use "Routing | Import" and expect everything from this process.
OsmAnd seems to handle this automatically. How we can improve this in Cruiser?

We could import GPX waypoints in a special favorites group? If not used in routing and probably as option.
This special group of favorites will only exist once, each import overwrites it, so there will be no duplicates.

**Reply by MotoUKRider** [reply]
Thinking about it, I have an issue with users *expecting* that workflow (Favorite import) from the Routing|Import function.  To me, that method should simply open a route/track/overlay for navigation.

Perhaps if we go down the temporary group "route", there will be an option to "Clear Temporary Favorites" when you clear the Route?  In that scenario, a user can deselect it and then open the Group to multi-select items to move into a normal Group - or clear them entirely when clearing the Route.

**Reply by devemux86** [reply]
> Perhaps if we go down the temporary group "route", there will be an option to "Clear Temporary Favorites" when you clear the Route?

As these waypoints are connected with the imported route, the simplest is to delete them with the route (like OsmAnd).
(if we want to store them forever, then it is better to import them as favorites)

**Reply by 0709wiwiwi** [reply]
The options you have with standard gpx track + waypoints.

A. You consider the waypoints as free located poi waypoints. 
Please keep them together with the track. (Delete track is delete poi waypoints).
- Overlay + routing (track).
After import with overlay you see them as dots, they are not announced (now).

B. You consider the waypoints as routing waypoints.
- Overlay + routing (wpt). Calculate the track from the waypoints. 
  You may still have to "adjust" (a lot) to "follow" the track overlay.


I don't use OsmAnd but OsmAnd possibly handles this as follows.

Track with (poi) waypoints.

- Shows the waypoints (default app icon) and announces them. 
- The import track is adapted by map matching (OsmAnd routable map).
  After this the (OsmAnd) turn instructions are generated
  
- Komoot (online) does this in a similar way, by the way.  (online)
  First map matching and then generate the instructions.

- Zumo users can test this ?
  By map matching can follow an import track and  attach instructions?

The map matching quality (map data) helps determine the result. 
There can definitely be variations compared to the import overlay.
So, as with any (re)calculation, there may be differences.
Just as my tax office also always calculates differently from me.

Translated with DeepL.com (free version)

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the summary.

Case B is already fully supported: Import GPX waypoints for routing.

We mean the case A: Import GPX track for routing.
Question: what to do with waypoints in the same GPX?

**Comment by Biker729** [comment]
TET-Routen sind Track GPX mir vielen Wegpunkten,
und ein Track fährt auch dort wo eine Route GPX nicht fährt.

**Reply by 0709wiwiwi** [reply]
@Biker729  Your request is:

- Keep the original track design intact.
- Announce the gpx wpt as text like a Via point.
- You don't say anything about turn instructions. 
- Do you want turn instructions or maybe not ?

**Reply by devemux86** [reply]
@0709wiwiwi The route is required and things are simple:
- Import GPX track to calculate a route
(this is the only way to have a route similar to the GPX path)

- Use GPX waypoints in navigation
(like the nearby favorites and POI in Cruiser navigation)

**Reply by 0709wiwiwi** [reply]
Yes that could fit. We are just going over the possibilities.  
As for that rather recurring question about the TET routes, users who really do ride them should still do more testing and report how the reroute.brf profile along with many shaping points produces useful results or not for them. @WolfgangZ42 is already testing this.

**Reply by WolfgangZ42** [reply]
To make it short: it works! But not always - I've seen situations with no valid OSM definitions ...

Here an example with positiv result:
Trying to make a route from a given offroad track (green dotted line). Up to now I had to use direct line for two segments:
![1](https://github.com/devemux86/cruiser/assets/121806238/796d3b15-1fe3-472b-80fd-8d26a6dadbf5)
Setting the profile to "reroute" and assigning this to WP22 shows the expected result:
![2](https://github.com/devemux86/cruiser/assets/121806238/f4e90f4b-f0f3-48ec-8feb-5f2c2e9e88f7)
For the next segment I needed some more shaping points, but finally it works too:
![3](https://github.com/devemux86/cruiser/assets/121806238/1e2a39e1-e203-41f9-8a4d-8584f07c5dcf)
In this case I'm sure, that the given track was been driven before - if you plan from scratch, you may encounter some surprises (staircases?).

I think, "reroute" will not be necessary very often - but it's now in my toolbox.

**Comment by Biker729** [comment]
@0709wiwiwi
Möchten Sie Abbiegeanweisungen oder nicht?

Ja, das meinte ich mit:

..mit Namen und Koordinaten entlang der Strecke angezeigt und angesagt werden.

Für Routen vorwiegend auf Straßen mit wenig Offroad sind Sprachansagen von Abbiegeanw. und WP sehr hilfreich.
Für Offroad ACT und TET wären  Abbiegeanweisungen mit Sprachansage nicht unbedingt nötig.

Einen Track 1:1 anzuzeigen war / ist mit Kurviger Web und App schon immer eine Schwachstelle,
einen Track 1:1 ohne Nachbesserungen anzuzeigen mit Sprachansage von WP und Abbiegungen wäre Top.

**Reply by WolfgangZ42** [reply]
Also ich benötige keine Namen bei Abbiegeanweisungen, und schon gar keine Koordinaten ("48.939420, 12.573110"?). Weder für Offroad Touren noch Onroad. Mir reicht "in 300 m rechts abbiegen". Und von WP will ich nichts hören/sehen, außer ich tippe gezielt darauf. Routenpunkte (das sind keine WP!) werden bereits perfekt angesagt.

Einen Track 1:1 anzuzeigen ist auch in Kurviger kein Problem (hier in gestricheltem rot). Daraus eine Route (blaue Linie) erstellen schon eher - siehe die Tour, die ich weiter oben zum Thema "reroute" als Beispiel genommen habe:
![image](https://github.com/devemux86/cruiser/assets/121806238/493a3028-424d-4816-b8f4-a57bb1a9dab8)


I don't need names for turn-by-turn directions, and certainly no coordinates ("48.939420, 12.573110"?). Neither for off-road tours nor on-road. "In 300m turn right" is enough for me. And I don't want to hear/see anything about WP unless I specifically tap on it. Route points (these are not WP!) are already perfectly announced.

Displaying a track 1:1 is also no problem in Kurviger (here in dashed red). Creating a route (blue line) from it is more of a problem - see the tour I used as an example above on the subject of "reroute".

**Reply by 0709wiwiwi** [reply]
Hi @WolfgangZ42 
Can you provide some examples of TET tracks?

**Reply by WolfgangZ42** [reply]
look here: https://github.com/devemux86/cruiser/discussions/531#discussioncomment-9389396
Do you need more?

**Reply by WolfgangZ42** [reply]
[Part of Day3 - ACT PT.gpx.txt](https://github.com/devemux86/cruiser/files/15282340/Part.of.Day3.-.ACT.PT.gpx.txt)

**Comment by Biker729** [comment]
Man kann sich ACT, TET-Track Touren auf verschiede Arten anzeigen lassen
mit einen grünen Strich auf dem Display oder einer Karte im Tankrucksack.

Da alle TET Routen mit vielen Infos-WP verfügbar sind wird das der Standart sein,
zumal man das mit 2 Klicks im Menü ändern kann:

![1](https://github.com/devemux86/cruiser/assets/146021011/d99604e2-e3ad-4f01-9553-c5a0feb2a7ba)
![2](https://github.com/devemux86/cruiser/assets/146021011/66bf6efa-5488-45c7-911b-bcdc00a31657)
![3](https://github.com/devemux86/cruiser/assets/146021011/132a4413-a44d-4ee2-bd7f-58c832aba596)
![4](https://github.com/devemux86/cruiser/assets/146021011/cc8b44ec-c3d3-4525-8d72-0a435116d10e)

Nun aber nochmal zurück zu meiner **eigentlichen Frage**:  
kann man mit Cruiser einen / mehrere „GPX Tracks sowie Zwischenziele“ **in einer GPX-Datei** speichern / importieren 
und mit dieser GPX ohne **Nacharbeiten** 1:1 dann beim Navigieren Sprachansagen von Zwischenzielen 
und Abbiegehinweisen erhalten.
Lösche ich diese GPX-Datei sind der GPX Track sowie die Zwischenziele wieder gelöscht.

**Reply by 0709wiwiwi** [reply]
@Biker729

Please attach the gpx file and mention the reference app.
TET Portugal contains 5 tracks and waypoints with Garmin sym.
App ? The Gpx inViewer app by the looks of it I suspect.

The Gpx Viewer shows each track with a separate colour.
Supports some Garmin sym by a png icon display.

Your question. 
Cruiser does sure not offer what the Gpx Viewer shows you thus.

**Reply by WolfgangZ42** [reply]
In den obigen screenshots ist zumindest die Dauer der einzelnen Tagesetappen falsch. 
Das TET-File "P.gpx" enthält sage und schreibe 200 WPT und 5 TRK Abschnitte. 
Nun könnten die WPT als Favoriten gespeichert werden - hier zwei screenshots dazu:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/156d6729-eeae-4397-91ac-8d14e83486ea"/>   <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/442e5a73-5c7a-4891-b80a-73453fca125e"/>
Die zeigen allerdings auch Tracks und Favoriten einer von mir geplanten Tour. 

Diese Favoriten kann man sich anzeigen und "vorlesen" lassen, wenn die Route nah genug daran vorbei führt. (_Navigation/Nearby favorites_ und _Voice guidance/Announcement/Nearby favorites_) Die Ansagen von Zwischenzielen und Abbiegehinweisen erfolgt ohnehin standardmäßig.

Einen TET-Track ohne Nacharbeiten in eine Route umzuwandeln ist nicht möglich, denn Teile dieser Tracks sind mitunter nicht einmal in OSM definiert. 

- - - - -
In the screenshots above, at least the duration of the individual daily stages is incorrect. 
The TET file "P.gpx" contains no less than 200 WPT and 5 TRK sections. 
Now the WPT could be saved as favorites - here are two screenshots: ...
However, they also show tracks and favorites of a tour I have planned. 

These favorites can be displayed and "read out" if the route passes close enough to them. (_Navigation/Nearby favorites_ and _Voice guidance/Announcement/Nearby favorites_) Intermediate destinations and turn-by-turn directions are announced by default anyway.

It is not possible to convert a TET track into a route without reworking it, as parts of these tracks are sometimes not even defined in OSM.

**Comment by Biker729** [comment]
@0709wiwiwi
Sie können die Screenshot von GPX Viewer ignorieren, 
was ich möchte ist eine Antwort auf meine einfache Frage die ich mit Thema #531 gestellt habe:

Kann man mit Cruiser einen / mehrere „GPX Tracks sowie Zwischenziele“ in einer GPX-Datei speichern / importieren
und mit dieser GPX ohne **Nacharbeiten 1:1** dann beim Navigieren Sprachansagen von Zwischenzielen
und Abbiegehinweisen erhalten.
Lösche ich diese GPX-Datei sind der GPX Track sowie die Zwischenziele wieder gelöscht.

Wird Cruiser das in Zukunft unterstützen. ?

Gruß und Danke

**Reply by WolfgangZ42** [reply]
Ich glaube, wir sind in diesem Thread hinreichend auf deine Fragestellung eingegangen - vielleicht liest du dir das alles nochmal durch. 

---
I think we have dealt with your question sufficiently in this thread - perhaps you should read it all again.

**Reply by devemux86** [reply]
@Biker729 

If you mean GPX with track + waypoints, it can be imported into Cruiser and calculate a route from the track.
Use the options as in the picture and the waypoints will also be imported.

If you want the GPX waypoints to be used in navigation, they currently need to be imported as favorites.
We discuss here how we can improve this.

---

Wenn Sie GPX mit Track + Wegpunkten meinen, kann es in Cruiser importiert werden und eine Route aus dem Track berechnen.
Verwenden Sie die Optionen wie auf dem Bild und die Wegpunkte werden auch importiert werden.

Wenn Sie möchten, dass die GPX-Wegpunkte in der Navigation verwendet werden, müssen sie derzeit als Favoriten importiert werden.
Wir diskutieren hier, wie wir dies verbessern können.

<img src="https://github.com/devemux86/cruiser/assets/121806238/2928b638-cc8f-4662-8814-b757e5ed4732" width="47%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c3e5c063-a698-4098-b158-5462efe7979a" width="53%"/>

**Comment by Biker729** [comment]
Vielen Dank für die Antworten, das Thema kann aus meiner Sicht geschlossen werden.

**Comment by 0709wiwiwi** [comment]
A simple Cruiser optimisation may be possible. 

Cruiser **NEW** suggestion
Convert a track to a navtrack to follow.
- By default add sym pass_place and type via at the start and finish trackpoint.
- Calculate by **0 route points** OR by +2 route points. (Set Slider range 0 # 2 to 200)  ( 0 is no calculation)

**Advantage**. 
You have a distance indicator to the finish.
Free colour and width of the line to follow. 
Get an offtrack warning. (Set no autorecalculation)
When offtrack, the direction arrow points to the finish.
Freely add announced info via's without a recalculation.

An example (manual notepad edit) is attached.
[TET Portugal -wpt_5_navtrk.gpx.txt](https://github.com/devemux86/cruiser/files/15307711/TET.Portugal.-wpt_5_navtrk.gpx.txt)

**Reply by devemux86** [reply]
@0709wiwiwi I do not understand if you are reporting any specific issue.
Please write more details about how to reproduce, if there is something to improve.

**Reply by 0709wiwiwi** [reply]
Report: The latest beta version 4 does NOT allow adding shaping or via points without calculation !)

_Cruiser_Brouter is not able to convert tracks that contain non routable track sections into a TBT navtrack.
RouteYou web converts single imported tracks  quite solidly  inclusive TBT on the routable tracksections._**

**Reply by devemux86** [reply]
> The latest beta version 4 does NOT allow adding shaping or via points without calculation

Add via / shaping points on the route without calculation: this works with single routes.
Not with alternative routes. As waypoints are shared and used by all alternative routes.

Basic routing principle: all calculated alternative routes have the same waypoints.

If using an editor to copy / paste unrelated routes in a file, they meant to be read only.
Note that users are not expected to use editors, they just want to use the app interface.

**Reply by devemux86** [reply]
Or can add via / shaping points on a route with alternatives without calculation,
but remove the alternatives, as the route structure has been radically changed.

**Reply by devemux86** [reply]
"Working" somewhere doesn't mean it doesn't cause complicated problems elsewhere. :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
_Consequently, the original TET P download I used contains multiple so (5) tracks into the same gpx file. 
With single track imports the suggestion would work.

**Reply by devemux86** [reply]
About the `?` in the instructions list, the gpx looks wrongly constructed?
Isn't it a navigation track?

It does not have at all the required `<sym>pass_place</sym>` tag.

**Reply by 0709wiwiwi** [reply]
The sym pass_place tag is necessary for Locus . Cruiser for a quick test did not need it.  In a final version should be attached.

**Reply by devemux86** [reply]
> Cruiser does not need it

Who said that? It is part of the offline GPX navigation track specification, so it is expected and needed!

**Reply by devemux86** [reply]
I think I found a way to make it work without calculation even when we have alternative routes.
You can test it in the next Beta.

**Reply by 0709wiwiwi** [reply]
> Without calculation even when we have alternative routes.

The function recalculate sections fails as it triggers a recalculate (all) route.

**Reply by devemux86** [reply]
> The function recalculate sections fails as it triggers a recalculate (all) route.

Intelligent features are too complex to work on routes not related to each other, that most users will never use.
(they can only be created with an editor)

"Recalculate section" works with "normal" route or routes:
- Single route
- Multiple alternatives with same waypoints
(like when all are calculated by the router)

**Reply by 0709wiwiwi** [reply]
Select just a single track is fine. Creating a no TBT navtrack is simple. 
It could so allow Cruiser to annouce the imported nearby **gpx waypoints.** 

Cruiser extra. 
**Navigation/Panels.**
- Nearby favorites.
- Nearby points of interest. 
- **Nearby gpx waypoints.** **= NEW** 
Display both the gpx (nav)track (line) and waypoint (icon) **PLUS announce the nearby waypoints.** 

It would be highly desirable that Cruiser supports the popular Garmin sym with the accompanying (png) icon.
![image](https://github.com/devemux86/cruiser/assets/54167201/023aee07-8fd9-44eb-bd2c-ca370e69a9a0)

**Reply by devemux86** [reply]
Delete "alternative" routes is already available and does not require calculation.

About Garmin icons, it requires work, there is a topic here:
- https://github.com/devemux86/cruiser/discussions/529

We have to see which features are most popular to implement in the next versions of Cruiser.
